rtos/TARGET_CORTEX/mbed_rtx_idle.cpp

```diff
@@ -193,8 +190,8 @@ static void default_idle_hook(void)
     uint32_t elapsed_ticks = 0;
 
     core_util_critical_section_enter();
-    uint32_t ticks_to_sleep = svcRtxKernelSuspend();
```

Comment:
That may not work, if we deprivilege the OS eg. for uVisor we are disabling the interrupts and then calling `osKernelSuspend` which will use SVC and that will fail.

Reply:
I recall now the use of this line, however we should not be using internal function, will check with CMSIS

Reply:
`svcRtxKernelSuspend` is internal API and not declared `extern` so it can't be called, directly. The crucial part is that all the functionality implemented inside `svcRtx*` functions must not be called with a different preemption level than assigned to SVC exception. Otherwise execution of this code is not longer guaranteed to be mutually exclusive. That could lead to control block corruption.
Why do you need interrupts to be disabled? If we can sketch the use case we might be able to relax interrupt restrictions for `osKernelSuspend` and `osKernelResume`.

Reply:
Actually I was saying some gibberish about deprivileging the os... that doesn't make any sense. As for the critical section, there was quite a long discussion that I can't recall now exactly. @c1728p9 should know